eidas_proxy_service/storage/ignite.py

```diff
@@ -0,0 +1,52 @@
+"""Storage for Light Requests and Responses backed by Apache Ignite."""
+from typing import Optional
+
+from pyignite import Client
+from pyignite.cache import Cache
+
+from eidas_proxy_service.models import LightRequest, LightResponse
+from eidas_proxy_service.storage import LightStorage
+from eidas_proxy_service.utils import dump_xml, parse_xml
+
+
+class IgniteStorage(LightStorage):
+    """
+    Apache Ignite storage for Light Requests and Responses.
+
+    :param host: Ignite service hostname or IP address.
+    :param port: Ignite service port.
+    :param request_cache_name: The cache where LightRequests are stored.
+    :param response_cache_name: The cache where LightResponses are stored.
+    """
+
+    def __init__(self, host: str, port: int, request_cache_name: str, response_cache_name: str):
+        self.host = host
+        self.port = port
+        self.request_cache_name = request_cache_name
+        self.response_cache_name = response_cache_name
+        self._client = None  # type: Optional[Client]
+
+    def get_cache(self, cache_name: str) -> Cache:
+        """Get an Ignite Cache."""
+        if self._client is None:
+            self._client = Client()
+            self._client.connect(self.host, self.port)
```

Comment:
A `try .. except` might be convenient, also it is always a good idea to set an explicit timeout. The default value is a blocking call without timeout :)

Reply:
I agree with the timeout. What do you mean by `try .. except`? Catching connection error?

Reply:
My impression was that this throws `OsError` and similar but that might have been for `put`/`get`...

Reply:
`Client.connect` raises `ConnectionError`, `put` and `get` also raise `pyignite.exceptions.CacheError` if the cache doesn't exist.

Anyway, do we want to catch these errors or let them trigger HTTP 500 so that admins get notified about the issue?

Reply:
We agreed not to catch the errors for now.